[BUG] Fix flaky org.opensearch.knn.index.codec.jvector.KNNJVectorTests.testJVectorKnnIndex_filter_maxInnerProduct test case

**What is the bug?**
The test case org.opensearch.knn.index.codec.jvector.KNNJVectorTests.testJVectorKnnIndex_filter_maxInnerProduct is flaky:

```
  2> java.lang.AssertionError: expected:<1> but was:<2>
        at __randomizedtesting.SeedInfo.seed([6BCA4ADF05D9CA6D:CF24E6FABC426A35]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.knn.index.codec.jvector.KNNJVectorTests.testJVectorKnnIndex_filter_maxInnerProduct(KNNJVectorTests.java:235)
```

**How can one reproduce the bug?**
```
./gradlew test --tests KNNJVectorTests.testJVectorKnnIndex_filter_maxInnerProduct -Dtests.seed=6BCA4ADF05D9CA6D
```

**What is the expected behavior?**
A clear and concise description of what you expected to happen.

**What is your host/environment?**
 - CI

**Do you have any screenshots?**
If applicable, add screenshots to help explain your problem.

**Do you have any additional context?**
- https://github.com/opensearch-project/opensearch-jvector/actions/runs/31636916959/job/94249506595?pr=679
- https://github.com/opensearch-project/opensearch-jvector/actions/runs/31636916988/job/94249494435?pr=679

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Enhancements
 
 ### Bug Fixes
+- Fix flaky testJVectorKnnIndex_filter_maxInnerProduct test case [681](https://github.com/opensearch-project/opensearch-jvector/pull/681)
 
 ### Infrastructure
 ### Documentation

--- a/src/test/java/org/opensearch/knn/index/codec/jvector/KNNJVectorTests.java
+++ b/src/test/java/org/opensearch/knn/index/codec/jvector/KNNJVectorTests.java
@@ -216,6 +216,7 @@ public class KNNJVectorTests extends LuceneTestCase {
         indexWriterConfig.setUseCompoundFile(false);
         indexWriterConfig.setCodec(getCodec(random().nextBoolean()));
         indexWriterConfig.setMergePolicy(new ForceMergesOnlyMergePolicy());
+        indexWriterConfig.setMaxBufferedDocs(totalNumberOfDocs);
         final Path indexPath = createTempDir();
         log.info("Index path: {}", indexPath);
         try (FSDirectory dir = FSDirectory.open(indexPath); IndexWriter w = new IndexWriter(dir, indexWriterConfig)) {
@@ -231,7 +232,8 @@ public class KNNJVectorTests extends LuceneTestCase {
             w.commit();
 
             try (IndexReader reader = DirectoryReader.open(w)) {
-                log.info("We should now have a single segment with 10 documents");
+                log.info("We should now have a single segment with {} documents", totalNumberOfDocs);
+
                 Assert.assertEquals(1, reader.getContext().leaves().size());
                 Assert.assertEquals(totalNumberOfDocs, reader.numDocs());
 

